src/pymor/discretizers/builtin/gui/jupyter/vista.py

```diff
@@ -0,0 +1,193 @@
+        my_theme.interactive = False
+    else:
+        my_theme.interactive = interactive if interactive is not None else True
+    my_theme.transparent_background = True
+    if is_jupyter():
+        # setting this to anything in non jupyter-contexts produces weird results
+        my_theme.jupyter_backend = 'ipygany'
+    my_theme.title = title
+    my_theme.axes.show = False
+    # apply it globally
+    pv.global_theme.load_theme(my_theme)
+    return my_theme
+
+
+def _get_default_bar_args():
+    # interactive=True currently triggers an Attribute error in
+    # pyvista.plotting.scalar_bars.ScalarBars.add_scalar_bar
+    return {'interactive': False}
+
+
+def visualize_vista_mesh(meshes, bounding_box=([0, 0], [1, 1]), codim=2, title=None, legend=None,
+                         separate_colorbars=False, rescale_colorbars=False, columns=2, color_map='viridis'):
+    from pyvista.utilities.fileio import from_meshio
+    from pyvistaqt import MultiPlotter
+
+    render_size = (300, 300)
+    theme = _load_default_theme()
+    scalar_bar_args = _get_default_bar_args()
+    if is_jupyter():
+        MultiPlotterType = _JupyterMultiPlotter
+    else:
+        MultiPlotterType = MultiPlotter
+    if isinstance(meshes, tuple):
+        rows = ceil(len(meshes)/2)
+        plotter = MultiPlotterType(nrows=rows, ncols=columns, window_size=render_size, theme=theme)
+        for meshlist, ind in zip(meshes, np.unravel_index(list(range(len(meshes))), shape=(rows, columns))):
+            plotter[ind].add_mesh(from_meshio(meshlist[0]), scalar_bar_args=scalar_bar_args)
+        plotter.link_views()
+    else:
+        plotter = pv.Plotter(theme=theme)
+        plotter.add_mesh(from_meshio(mesh=meshes[0]), scalar_bar_args=scalar_bar_args)
+    plotter.camera_position = 'xy'
+    return plotter.show()
+
+
+class PyVistaPatchWidget(QtInteractor):
+
+    def __init__(self, U, limits, parent, grid, bounding_box=([0, 0], [1, 1]), codim=2):
+        from qtpy.QtWidgets import QSizePolicy
+        assert grid.reference_element in (triangle, square)
+        assert grid.dim == 2
+        assert codim in (0, 2)
+        theme = _load_default_theme(interactive=(grid.dim == 3))
+        super().__init__(parent, theme=theme)
+        self.setMinimumSize(300, 300)
+        self.setSizePolicy(QSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding))
+
+        self.scalar_name = 'Data'
+        self.mesh_kwargs = {'scalar_bar_args': _get_default_bar_args(),
+                            'scalars': self.scalar_name}
+        self.grid = grid
+        self.reference_element = grid.reference_element
+        self.codim = codim
+        self.set(U, limits)
+        if grid.dim != 3:
+            # disable interactive camera
+            self.disable()
+
+    def set(self, U, limits):
+        self.clear()
+        self.limits = limits
+        self.mesh_data = to_meshio(self.grid, data=U, scalar_name=self.scalar_name, codim=self.codim)
```

Comment:
This is clearly not optimal compared to our old approach. There is no need to create a new mesh, we only have to update the data. In the old code, I believe, we stored a fixed mapping from vertices to dofs in order to quickly obtain the data associated with the mesh.